eth/tracers/native/prestate.go

```diff
@@ -135,12 +166,64 @@ func (t *prestateTracer) CaptureTxStart(gasLimit uint64) {
 	t.gasLimit = gasLimit
 }
 
-func (t *prestateTracer) CaptureTxEnd(restGas uint64) {}
+func (t *prestateTracer) CaptureTxEnd(restGas uint64) {
+	if !t.config.CollectPost {
+		return
+	}
+
+	for addr, state := range t.pre {
+		// the deleted account's state is pruned
+		if _, ok := t.deleted[addr]; ok {
+			continue
+		}
+		modified := false
+		postAccount := &account{Storage: make(map[common.Hash]common.Hash)}
+		newBalance := bigToHex(t.env.StateDB.GetBalance(addr))
+		newNonce := t.env.StateDB.GetNonce(addr)
+		newCode := bytesToHex(t.env.StateDB.GetCode(addr))
+
+		if newBalance != t.pre[addr].Balance {
+			modified = true
+			postAccount.Balance = newBalance
+		}
+		if newNonce != t.pre[addr].Nonce {
+			modified = true
+			postAccount.Nonce = newNonce
+		}
+		if newCode != t.pre[addr].Code {
+			modified = true
+			postAccount.Code = newCode
+		}
+
+		for key, val := range state.Storage {
+			newVal := t.env.StateDB.GetState(addr, key)
+			if val != newVal {
+				modified = true
+				postAccount.Storage[key] = newVal
+			}
+		}
+		if modified {
+			t.post[addr] = postAccount
+		}
+	}
+	for a := range t.created {
+		delete(t.pre, a)
```

Comment:
I'm not sure I understand this part. Why did you have to delete them explicitly?

Reply:
These newly created contracts, which were not previously on the statedb, so need to be deleted.

Reply:
Aha there's something we have to check. It might be that the current `prestateTracer` also includes the address at which a new contract/eoa is being created. It might be an empty account, or not (have existing balance). In this case this becomes a breaking change.

Reply:
Sounds reasonable, Alice send some Ether to Bob(not exist yeat), then Alice created the contract(whose address is Bob).

Reply:
So let's delete those contract addresses that don't exist(no balance, no code and no storage) in the previous state.

Reply:
@s1na Seems the original `prestateTracer` also has this issue, it will delete the `tx.to` 


https://github.com/ethereum/go-ethereum/blob/69568c554880b3567bace64f8848ff1be27d084d/eth/tracers/native/prestate.go#L85-L89

So, in the case we discussed before, we can't fetch the full prestate. 

eg: this address [0x332b656504f4eabb44c8617a42af37461a34e9dc](https://etherscan.io/address/0x332b656504f4eabb44c8617a42af37461a34e9dc). If we `prestateTrace` the [tx 0x9aac76c9ed191da631b5e70c8824f2ad106a0f5147d57c611ed64653734dd85f](https://etherscan.io/tx/0x9aac76c9ed191da631b5e70c8824f2ad106a0f5147d57c611ed64653734dd85f):

> request
```
{
    "jsonrpc": "2.0",
    "method": "debug_traceTransaction",
    "params": [
        "0x9aac76c9ed191da631b5e70c8824f2ad106a0f5147d57c611ed64653734dd85f",
        {
            "tracer": "prestateTracer",
            "timeout": "120s",
            "tracerConfig": {
                "collectPost": false
            }
        }
    ],
    "id": 67
}
```

> response 

```
{
    "jsonrpc": "2.0",
    "id": 67,
    "result": {
        "0x082d4cdf07f386ffa9258f52a5c49db4ac321ec6": {
            "balance": "0xc820f93200f4000",
            "nonce": 94,
            "code": "0x"
        },
        "0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5": {
            "balance": "0xbf681825be002ac452",
            "nonce": 28922,
            "code": "0x"
        },
        "0x82effbaaaf28614e55b2ba440fb198e0e5789b0f": {
            "balance": "0xb3d0ac5cb94df6f6b0",
            "nonce": 1,
            "code": "0x"
        }
    }
}
```

missing the 5305ETH of contract 0x332b656504f4eabb44c8617a42af37461a34e9dc

So, do you think we need to fix it or don't worry？

Reply:
Aah good find! up to you, if you want i can create an issue to fix it later.

Reply:
thanks, I'll file a new PR after this one merged.